### ISSUE TITLE
[java] UseStringBufferLength: false negative with empty string variable

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3639]
**Affects PMD Version:6.40.0** 

**Rule:** [UseStringBufferLength](https://pmd.github.io/latest/pmd_rules_java_performance.html#usestringbufferlength)

**Description:**
I think PMD should have reported a violation at line 4, but it doesn't. This code sample can reduce the performance, so I think it is a false negative. We should consider constant String for this rule. 🤔
 
**Code Sample demonstrating the issue:**

```java
public void foo() {
    StringBuffer sb = new StringBuffer("any_string");
    final String nullStr = "";
    if (sb.toString().equals(nullStr)) { // PMD should report a warning here
        System.out.println("Buffer is empty");
    }

    // the preferred way
    if (sb.length() == 0) {
        System.out.println("Buffer is empty");
    }
}
```

**Expected outcome:A warning**

PMD should report a violation at line 4, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*
